CONTRIBUTING.md

```diff
@@ -61,6 +61,39 @@ This basically builds `terracognita` with the current sources.
+
+**Resources**
+
+Resources are each of one of the Terraform resources that this specific provider has and can import using TerraCognita. The main idea is that the `provider.Provider.Resources` returns all the instances of one resource. That information has to be fetched from the provider API using the sdk initialized before and for each instance we have to build the Terraform resource ID and return the resource. This then will be used by Terraform to read all the needed data from the provider to load the resource Schema (no need to do anything as this is abstracted into the caller of that function).
+
+To know what's the ID expected by Terraform you can check on the Terraform documentation of the resource ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#import)[TODO: link docs and tf code], sometimes is more complex and it needs several elements to build an ID ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy#import) and [code](https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_iam_role_policy.go#L171)) so we have to set it on the right format for then be read for Terraform.
+
+As providers have a lot of resources we have created custom sdks on top of the provider sdks that we use to simplify things. Those wrappers are autogenerated from a configuration that is custom for each Provider. We do this so when we have to add new resources is easy, one of the main reasons is to also avoid pagination and always return all the resources directly. For example in `aws` we have the [configuration](https://github.com/cycloidio/terracognita/blob/master/aws/cmd/functions.go) which has the [go:generate](https://github.com/cycloidio/terracognita/blob/master/aws/reader/connector.go#L56) that generates the end [reader](https://github.com/cycloidio/terracognita/blob/master/aws/reader/reader.go). We have similar things for the other sdks
+
+The next point on the documentation explains how to add new resources to a provider which will help on doing it from 0 too.
+
+*Note:* If a provider does not have the `Import` function on that specific resource (meaning it cannot be imported by `terraform import ...` it'll not be imported. But we can dynamically add that function so it can be imported as on [this example](https://github.com/cycloidio/terracognita/blob/dae282137c905c90602ae7e0d82cc7601ff1fc85/aws/resources.go#L1629)
+
+**CMD**
+
+On the `cmd/` pkg you'll need to add the provider subcommand which you can mainly check how the other providers are done but it requires the initialization of the provider and the call to the main function `provider.Import()` with all the required parameters.
+
+Aside from that you also need to add the `resources` subcommand which will list all the resources supported by the provider in TerraCognita
 
```

Comment:
Perhaps add a section for documentation: where to add that a new provider is supported, the version, the list of supported resources, etc.

Reply:
Yep you are right